include/sisl/cache/simple_cache.hpp

```diff
@@ -42,14 +43,27 @@ class SimpleCache {
             m_key_extract_cb{std::move(extract_cb)},
             m_map{num_buckets, m_key_extract_cb, std::bind(&SimpleCache< K, V >::on_hash_operation, this, _1, _2, _3)},
             m_per_value_size{per_val_size} {
-        m_record_family_id = m_evictor->register_record_family(std::move(evict_cb));
+        m_record_family_id = m_evictor->register_record_family(std::move(evict_cb), [this](const CacheRecord& record) {
+            V const value = reinterpret_cast<SingleEntryHashNode< V >*>(const_cast< CacheRecord* >(&record))->m_value;
+            K key = m_key_extract_cb(value);
+            t_evicted_keys.insert(key);
+        });
     }
 
     ~SimpleCache() { m_evictor->unregister_record_family(m_record_family_id); }
 
     bool insert(const V& value) {
         K k = m_key_extract_cb(value);
-        return m_map.insert(k, value);
+        bool ret = m_map.insert(k, value);
```

Comment:
if a thread remove a key , and then insert it again , the removal and insertion happen in the same thread.  then , line 62 will remove the key just inserted.

should we add a check here to see if the key being inserted is already in the t_evicted_keys. if yes, remove it from t_evicted_keys

Reply:
`t_evicted_keys` is only populated during an insert and if eviction is needed. During a remove call, the `t_evicted_keys` should always be empty.